src/librustc_driver/driver.rs

```diff
@@ -954,8 +954,11 @@ pub fn build_output_filenames(input: &Input,
             if *odir != None {
                 sess.warn("ignoring --out-dir flag due to -o flag.");
             }
+
+            let cur_dir = Path::new(".");
```

Comment:
This default is actually more appropriately the empty string due to the way that normalization works currently.